core/gen/gen_types.go

```diff
@@ -22,34 +22,26 @@ package core
 
 var importFmt string = `
 import (
-	"fmt"
 	"io"
 )
 `
 
 var ensureObjectIsTemplate string = `
 func EnsureObjectIs{{.Name}}(obj Object, pattern string) {{.TypeName}} {
-	switch c := obj.(type) {
-	case {{.TypeName}}:
+	if c, yes := obj.({{.TypeName}}); yes {
 		return c
-	default:
-		if pattern == "" {
-			pattern = "%s"
-		}
-		msg := fmt.Sprintf("Expected %s, got %s", "{{.ShowName}}", obj.GetType().ToString(false))
-		panic(RT.NewError(fmt.Sprintf(pattern, msg)))
 	}
+	panic(FailObject(obj, "{{.TypeName}}", pattern))
```

Comment:
Should not we pass `"{{.ShowName}}"` here instead of `"{{.TypeName}}"` to preserve current behavior?

Reply:
Ah, indeed, I would think so! Will confirm and update as appropriate.